### PR TITLE
New/following pages

### PR DESCRIPTION
**Pull request by AndriiPopov3**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: backend/src/services/page.service.ts [hunk 1]
```diff
@@ -97,3 +97,54 @@ export const getPage = async (
...
+  const pageContentRepository = getCustomRepository(PageContentRepository);
+  const { followingPages } = await userRepository.findById(userId);
+  if (followingPages.length > 0) {
+    const pages = await Promise.all(followingPages.map(async page => {
+      const content = await pageContentRepository.findByPageId(page.id);
+      return {
+        id: page.id,
+        title: content.title,
+      };
+    }));
```
Comment (aleksandr-kuharenko) [inline backend/src/services/page.service.ts]: It is not the optimal approach. Please join PageContent to `followingPages` 
[example](https://github.com/typeorm/typeorm/issues/1270#issuecomment-348355428)

File: backend/src/data/repositories/page.repository.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { Page } from '../entities/page';
 @EntityRepository(Page)
 class PageRepository extends Repository<Page> {
   public findById(id: string): Promise<Page> {
-    return this.findOne({ id });
+    return this.findOne({ relations: ['followingUsers'], where: { id: id } });
```
Comment (aleksandr-kuharenko) [inline backend/src/data/repositories/page.repository.ts]:
```suggestion
    return this.findOne({ relations: ['followingUsers'], where: { id } });
```

File: frontend/src/components/pages/components/page-content/page-content.tsx [hunk 3]
```diff
@@ -12,10 +14,13 @@ import { pagesActions } from 'store/pages';
 import isUUID from 'is-uuid';
 import { useHistory } from 'react-router';
 import { AppRoute } from 'common/enums/enums';
+import { PageApi } from 'services';
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/pages/components/page-content/page-content.tsx]: We already call action to get page by id in this component. Why we cannot add information about page followers to existing request? We do not need extra requests

File: frontend/src/components/profile-info/profile-info.tsx [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
 import Row from 'react-bootstrap/Row';
 import Col from 'react-bootstrap/Col';
 import Card from 'react-bootstrap/Card';
 import ListGroup from 'react-bootstrap/ListGroup';
 import Badge from 'react-bootstrap/Badge';
-import { Link } from 'react-router-dom';
+import { Link } from 'components/common/common';
 import './profile-info.scss';
-import { UserApi } from 'services';
+import { UserApi, PageApi } from 'services';
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/profile-info/profile-info.tsx]: The same here. If we already call UserAPI, can we add info about the followed pages to the response?

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

**Review by ursarik — APPROVED** [review]

File: frontend/src/store/pages/actions.ts [hunk 5]
```diff
@@ -40,14 +42,29 @@ const getPage = createAsyncThunk(
...
+  },
+);
+
+const setCurrentPageFollowed = createAsyncThunk(
+  ActionType.SET_CURRENT_PAGE_FOLLOWED,
+  async (payload: boolean, { dispatch }) => {
+    dispatch(actions.setCurrentPageFollowed(payload));
+  },
+);
+
```
Comment (aleksandr-kuharenko) [inline frontend/src/store/pages/actions.ts]: you do not need async thunk here. You perform the synchronous operation.
Use action created in slice

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

**Pull request merged**